frontend/components/GithubAuthButton.jsx

```diff
@@ -16,8 +16,8 @@ const calcWindow = () => ({
   top: window.screen.height / 2 - 800 / 2,
 });
 
-function authorize() {
-  return new Promise((resolve, reject) => {
+function authorize(revokeFirst) {
```

Comment:
Is there anytime we wouldn't revoke first?

Reply:
On the main page, if they aren't connected to Github already, we don't really need to revoke first (although doing so won't cause issues as we catch the GH 404 in our own API). Fine to implement either way